### ISSUE TITLE
Feature Request: How to forward some semi-dynamic requests while blocking network access

### ISSUE THREAD

**Issue by hamlim** [ballercat/jambox#42]
## Background:

We ran into an interesting limitation the other day with our Jambox and Next.js setup, and wanted to get some insight / propose some options for what we think could unblock us.

The Next.js setup involves running two processes: An outer-shell reverse proxy that is a plain old express application (runs some middleware, fetches some data, etc) and then proxies to the second process which is Next itself. The express layer is running on port 3000 and the Next layer is running on port 3001.

Our Jambox config already forwards page requests to the underlying server running on port 3000, however we first run into an error about the proxied request to `localhost:3001` not being cached.
We can add `localhost:3001` to our `forward` config in the `jambox.config.js` file to get past this. However - Next is being a pain and running more servers behind the scenes to manage rendering the experiences (either via RSC or regular pages), and importantly these other servers are running on ports that are unknown to us. 

If I launch the experience, and grab the localhost port that the underlying internal Next server is launching and put that in our `forward` config - it seems to start working - however, restarting the Next process will cause it to launch the process on a different port the next time so it's not consistently using that same port.

## Ideas:

We were wondering if we could possibly support either:

- Looser matches on the keys in the `forward` config (in our case, dropping the port), but still forward to the same port based on the `target` config

e.g.

```js
module.exports = {
  forward: {
    'some-site.com': {
      target: 'localhost:3000'
    },
    // match requests for localhost with any port
    // actual syntax could change
    'localhost:{*}': {
      // RegEx-like syntax to use the capture groups in the `forward`ed origin
      target: 'localhost:{$1}'
    }
  }
}
```

- _Or_, like #34, support a `function` for the `forward` config, which could allow for a lot more dynamic behavior if desired

**Comment by hamlim** [comment]
I'll work on creating a simplified reproduction of the issue also!

**Comment by ballercat** [comment]
Fixed in `0.1.1`
